src/app/ui/cli.js

```diff
@@ -134,7 +158,37 @@ module.exports = class CLI {
       ? "open"
       : ""
 
+    // TODO: This does NOT seem ideal, but the idea is to force the timer bar `<div>` to re-kick-off the CSS animations when the `style` attribute changes. That can be done by forcing the browser to redraw (since Mithril itself doesn't redraw the whole element, it just updates the attributes).
```

Comment:
[onupdate](https://mithril.js.org/lifecycle-methods.html#onupdate) might be more appropriate eventually, once we move these to their own components.  I know you've said you're not happy with this current implementation, but it's relatively straight-forward and it is entirely implemented in vanilla.js which means it's easy to reason about.